core/src/main/java/org/testcontainers/utility/RegistryAuthLocator.java

```diff
@@ -206,7 +209,26 @@ private AuthConfig runCredentialProvider(String hostName, String credHelper) thr
                 .execute()
                 .outputUTF8()
                 .trim();
-        } catch (Exception e) {
+        }
+        catch (InvalidResultException e) {
+            if(e.getResult() != null && e.getResult().hasOutput()) {
+                // ErrCredentialsNotFound standardizes the not found error, so every helper returns the same message
+                // We can handle it properly with fallback to other resources.
+                // https://github.com/docker/docker-credential-helpers/blob/19b711cc92fbaa47533646fa8adb457d199c99e1/credentials/error.go#L4-L6
+                if ("credentials not found in native keychain".equals(e.getResult().outputString().trim())) {
+                    log.debug("Credentials not found in native keychain, credential helper ({}), return null to allow fallback", credentialHelperName);
+                    return null;
+                }
+                log.debug("Failure running docker credential helper ({}) with output '{}'",
```

Comment:
It's cool that we're now detecting the 'credentials not found' case and dealing with it in a better way!

Now, I wonder if the logging on line 222 and 226 should be at a higher level. There's probably no 'good' reason why these lines should be hit, so perhaps we should log at ~warn level to aid users in diagnosing a fault.

Reply:
I'm not sure, cause we are throwing exception from there, so maybe we should log it on higher level, like here:
https://github.com/testcontainers/testcontainers-java/blob/665e08bdbce7839a709badbc9c772fa11ec9fdce/core/src/main/java/org/testcontainers/utility/RegistryAuthLocator.java#L118-L123)